### ISSUE TITLE
sddm Wayland resolution problem on multiple screen with different resolution

### ISSUE THREAD

**Issue by miyoku157** [NixOS/nixpkgs#294260]
### Describe the bug
On my Nixos configuration, I tried to enable the option 
`Services.xserver.displayManager.Sddm.wayland.enable=true`
I'm using plasma 6 and hyprland which are both Wayland.
I've two monitor , one 1920x1080@60 and one 2560x1440@144.
And when Wayland Is enable I have a cropped sddm which doesn't allow me to choose my desktop.
I tried to disconnect one of the screen by executing a script which turn of one screen thanks to xrandr and the option:
`Services.xserver.displayManager.setupCommands`
But the problem is still here (but one of my screen is correctly disconnected)

### Expected behavior
Sddm printed correctly on both screen and getting access to the list of desktop manager
### Screenshots
![IMG_20240308_125538](https://github.com/NixOS/nixpkgs/assets/11454487/96d49749-e54b-4925-9204-b699248e0961)

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
 - host os: `Linux 6.7.7-zen1, NixOS, 24.05 (Uakari), 24.05.20240306.9df3e30`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/store/gig8j85kj7ybjy3ksn6k3aich8j2k59y-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by js6pak** [comment]
Temporary solution until #295839 is merged:
```nix
services.xserver.displayManager.sddm = {
  enable = true;

  wayland = {
    enable = true;
    compositorCommand = "${lib.getExe' pkgs.kdePackages.kwin "kwin_wayland"} --drm --no-lockscreen --no-global-shortcuts --locale1";
  };

  extraPackages = with pkgs; [
    kdePackages.layer-shell-qt
  ];

  settings = {
    General = {
      GreeterEnvironment = lib.concatStringsSep "," [
        "QT_WAYLAND_SHELL_INTEGRATION=layer-shell"
      ];
    };

    Theme = {
      # Both of these are nessecary otherwise the cursor isn't shown at all
      CursorTheme = "breeze_cursors";
      CursorSize = 24;
    };
  };
};
```

You probably also want to copy the display config using [the manual method described on archwiki](https://wiki.archlinux.org/title/SDDM#Match_KDE_display_configuration).

**Comment by K900** [comment]
Is this still an issue with kwin_wayland?

**Comment by miyoku157** [comment]
Hi, i try again today and it's still not working for me.
I upgraded my flake to use kwin to test, kwin work but i can't figured how to customize the background or setup. Even through theme like "breeze" for example
Even though the problem is still present if i use Weston 
Also, another thing i've noticed (But maybe unrelated) is that the config:
`autoNumlock=true;`
Doesn't work anymore, at boot it's never ON.

**Comment by Svenum** [comment]
I have the numlock problem too.

**Comment by peterhoeg** [comment]
For the purpose of this issue - with `kwin`, the resolution problem has been fixed, is that correct?

**Comment by miyoku157** [comment]
Yes, Kwin allow me to get both screen at good resolution and manage my
display. (But it's not as fancy as weston but it's another problem ^^)

Le lun. 25 mars 2024 à 13:21, Peter Hoeg ***@***.***> a
écrit :
